src/D2L.CodeStyle.Analyzers/Language/ParamNamesShouldMatchInterfaceAnalyzer.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.Analyzers.Language {
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	internal sealed class ParamNamesShouldMatchInterfaceAnalyzer : DiagnosticAnalyzer {
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+			Diagnostics.InterfaceImplementationParamNameMismatch
+		);
+
+		public override void Initialize( AnalysisContext context ) {
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction( RegisterAnalyzer );
+		}
+
+		private static void RegisterAnalyzer( CompilationStartAnalysisContext context ) {
+			context.RegisterSymbolAction(
+				ctx => AnalyzeClassDefinition(
+					ctx,
+					(INamedTypeSymbol)ctx.Symbol
+				),
+				SymbolKind.NamedType

```

Comment:
IRL @omsmith mentioned this came from the sealed analyzer. Maybe looking at `ClassDeclarationSyntax` instead would be good (and seems like it would apply to the sealed analyzer too...)

Downsides to named type:

- non-class things
- external symbols: we shouldn't bother checking these
- duplicate instances of a named type (would be fixed by caching?)

still would want to cache, maybe, due to `partial` classes (wouldn't want to emit diagnostics multiple times.)